deployment documentation (#159)
peridio/peridio-cli: docs/deployment.md

# Creating a CLI release

We manage CLI releases with github tags and github actions

## Update Cargo.toml

> Update the `Cargo.toml` version field, taking into account:
> 
> - minor patches are small fixes without new features or noticeable changes 
> - mayor patched are feature implementations or changes 

Reviewer: ```suggestion
Update the `Cargo.toml` version according to semantic versioning
```

Save the file and make sure to run `cargo build` for `Cargo.lock` to update 

> Push the changes with the commit message `stage 0.xx.xx release`, the commit message is not a requirement but we do it for consistency

Reviewer: ```suggestion
Push the changes with the commit message `stage x.y.z release`, the commit message is not a requirement but we do it for consistency
```

## Add a new tag 

Once the Cargo files changes are pushed and merged into main

- make sure to be in `main` and with the latest Cargo changes pulled 

> - run `git tag 0.xx.xx` which `0.xx.xx` should match your Cargo changes

Reviewer: ```suggestion
- run `git tag 0.xx.xx` which `x.y.z` should match your Cargo changes
```

> - run `git tag 0.xx.xx` which `0.xx.xx` should match your Cargo changes
> - run `git push origin 0.xx.xx` which `0.xx.xx` should match your Cargo changes

Reviewer: ```suggestion
- run `git tag x.y.z` which `x.y.z` should match your Cargo changes
- run `git push origin x.y.z` which `x.y.z` should match your Cargo changes
```

> Push the changes with the commit message `stage 0.xx.xx release`, the commit message is not a requirement but we do it for consistency
> 
> ## Add a new tag 
> 
> Once the Cargo files changes are pushed and merged into main
> 
> - make sure to be in `main` and with the latest Cargo changes pulled 
> - run `git tag 0.xx.xx` which `0.xx.xx` should match your Cargo changes
> - run `git push origin 0.xx.xx` which `0.xx.xx` should match your Cargo changes

Reviewer: ```suggestion
Push the changes with the commit message `stage x.y.z release`, the commit message is not a requirement but we do it for consistency

## Add a new tag 

Once the Cargo files changes are pushed and merged into main

- make sure to be in `main` and with the latest Cargo changes pulled 
- run `git tag x.y.z` which `x.y.z` should match your Cargo changes
- run `git push origin x.y.z` which `x.y.z` should match your Cargo changes
```
